README.adoc

```diff
@@ -1,5 +1,174 @@
+
+=== GeneratorModel
+
+The `GeneratorModel` Java class loads the Handlebars files from the base directory and overrides them with the files from the override directory if it is necessary.
+The class contains a global permission parameter that adds a global permission to all of the Handlebars files.
+(The `GeneratorTemplate` Java class, contained within the GeneratorModel class, includes all of the YAML representations listed earlier. The GeneratorModel maps the YAML properties to the GeneratorTemplate.)
+
+=== Generator Maven Project
+
+https://github.com/BlackBeltTechnology/judo-meta-esm/tree/develop/generator-maven-plugin#readme[This] link contains the description of the generator maven plugin.
+
+=== Helper registration
+
+We don't register helpers manually. To register a java helper class use the `@TemplateHelper` annotation.
+
+IMPORTANT: The Java class method must be static.
+
+If you inherit your helper class from the `StaticMethodValueResolver` class, you can refer to its methods using dot notation.
+
+The helpers are available in springEL too with # prefix.
+
+=== Self meaning in different situations
+
+The self mean different thinks in the springEL and Handlebars depend on the YAML properties:
+
+- If actorTypeBased is enabled, then #self refers to the actor.
+
+- If we have a `factoryExpression`, then #self refers to the result of the factory expression.
+
+- If we do not have a `factoryExpression`, then #self refers to the model.
+
+=== How to reach templateParameters
+
+The problem with the templateParameters is that it is not static. This can cause issues with parallel building. To void
+this issue the `@ContextAccessor` annotation and a `bindContext` method.
+
+Example:
+[source,java]
+----
+@TemplateHelper
+@ContextAccessor
+public class StoredVariableHelper extends StaticMethodValueResolver {
+
+    public static void bindContext(Map<String, ?> context) {
+        ThreadLocalContextHolder.bindContext(context);
+    }
+
+    public static synchronized String getApiPrefixLocal(Object object) {
+        return (String) ThreadLocalContextHolder.getVariable("apiPrefix");
+    }
+
+    public static synchronized Boolean isGenerateOptionalTypes(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generateOptionalTypes"));
+    }
+
+    public static synchronized Boolean isGeneratePayloadValidator(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generatePayloadValidator"));
+    }
+}
+
+----
+
+This only need if we used the templateParameters in a java Helper.
+
+=== Checksum generation
+A checksum is a value calculated from data to detect errors during transmission or storage. It ensures data integrity and any mismatches indicate errors in the data.
+
+* The checksum generation is located in the generator-maven-plugin.
```

Comment:
Its not true. It is called within the engine, but there is a seprated maven goal for that function. It is neccesary because sometimes the generated code is formatted which is from the checksum standpoint a modification. So that helps to "update" checksum for the new state.